src/stores/movies.ts

```diff
@@ -22,20 +22,24 @@ export const useMovieStore = defineStore({
         ...new Set(this.movieList.map((movie) => movie.genre.name)),
       ];
     },
+    titleList: (state) => {
+      return ["All movies", ...state.movieList.map((movie) => movie.title)];
+    },
```

Comment:
Both `genreList` and `titleList` are holding value for "All movies/categories" option which is used only for BaseSelect - if I'd like to use it somewhere else I'd have to create new getter. Let's simplify store and add those "All" options directly in components that are using this value.

F.ex:
```
titleList: (state) => state.movieList.map((movie) => movie.title);
```

`src/views/MoviesPage.vue`

```
const selectOptions = computed(() => ['All movies', ...moviesStore.genreList]);
```

Reply:
Point taken - make the code more modular not single purpose as it would create complications down the line - thank you